rhino/src/main/java/org/mozilla/classfile/ClassFileWriter.java

```diff
@@ -4471,11 +4516,25 @@ public String toString() {
         }
     }
 
+    @SuppressWarnings("AndroidJdkLibsChecker")
+    private static void printOrigin() {
+        StackWalker walker = StackWalker.getInstance(Option.RETAIN_CLASS_REFERENCE);
+        var callerFrame =
+                walker.walk(
+                        s ->
+                                s.filter(x -> x.getDeclaringClass() != ClassFileWriter.class)
+                                        .findFirst());
+        var f = callerFrame.orElseThrow();
+        System.err.printf("%s:%d ", f.getDeclaringClass().getSimpleName(), f.getLineNumber());
+    }
+
     private static final int FileHeaderConstant = 0xCAFEBABE;
     // Set DEBUG flags to true to get better checking and progress info.
     private static final boolean DEBUGSTACK = false;
     private static final boolean DEBUGLABELS = false;
     private static final boolean DEBUGCODE = false;
+    private static final boolean DEBUGCODEORIGINS = false;
+    private static final boolean DEBUGMETHODS = false;
```

Comment:
FYI, with #1722, I've added a config feature.
So you might add a config switch like this:
```suggestion
    private static final boolean DEBUGMETHODS = RhinoConfig.get("rhino.debug.methods", false)
```